rfcs/function-coroutine-close.md

```diff
@@ -0,0 +1,32 @@
+# coroutine.close
+
+## Summary
+
+Add `coroutine.close` function from Lua 5.4 that takes a suspended coroutine and makes it "dead" (non-runnable).
```

Comment:
Will dead threads be accepted? This exists in Lua 5.4 to close to be closed variables from threads that errored.

Will coroutine.close destroy the call stack of the thread, preventing use of debugging functions to inspect the call stack after coroutine.close is called? Lua 5.4 destroys the call stack when closing to be closed variables.

Since these functions can kill a suspended thread, will built in functions check to make sure they can resume the thread?
```lua
local t = coroutine.create(function()
task.wait()
print(1)
end)
coroutine.resume(t) -- t is waiting to be resumed
coroutine.close(t) -- t is dead, resuming it will now error
-- ... will an error be generated for attempting to resume a dead thread?
```

Reply:
> Will dead threads be accepted?

The RFC currently states that they will be. I think it makes sense to still do this because closing a coroutine clears out the stack, thus freeing a bunch of resources immediately even if the coroutine object is still reachable.

> Will coroutine.close destroy the call stack of the thread,

Yeah, that's the plan. This is somewhat necessary mechanically, as right now coroutine "dead" status is equivalent to "call stack is empty".

> Since these functions can kill a suspended thread, will built in functions check to make sure they can resume the thread?

Yeah, that already happens - if a coroutine errors or runs to completion, you can't resume it anymore. `close` will just transition the thread to the same state.

Reply:
I've updated the RFC to clarify the second and third point.